Sources/mpc-core-kit-swift/MpcCoreKitSwift.swift

```diff
@@ -15,129 +15,113 @@ import TorusUtils
+    // internal var state: CoreKitstate
 
-    internal var appState: CoreKitAppState
+    public var metadataPubKey: String?
+    public var oAuthKey: String?
+    // share index used for backup share recovery
+    public var deviceMetadataShareIndex: String?
+    public var loginTime: Date?
 
     public var metadataHostUrl: String?
-
     public var tkey: ThresholdKey?
-
     public var tssEndpoints: [String]?
     public var authSigs: [String]?
     public var verifier: String?
     public var verifierId: String?
-
     public var torusUtils: TorusUtils
     public var nodeIndexes: [Int]?
     public var nodeDetails: AllNodeDetailsModel?
-
     public var nodeDetailsManager: NodeDetailManager
-
     public var sigs: [String]?
 
     public var coreKitStorage: CoreKitStorage
-
     private let storeKey = "corekitStore"
+    private let localstateKey = "localstate"
+    private let customAuth: CustomAuth
 
-    private let localAppStateKey = "localAppState"
-
-    // init
-    // TODO: This should accept a param class instead
-    public init(web3AuthClientId: String, web3AuthNetwork: Web3AuthNetwork, disableHashFactor: Bool = false, localStorage: ILocalStorage, manualSync: Bool = false) throws {
-        if web3AuthClientId.isEmpty {
+    public init(options: CoreKitWeb3AuthOptions) throws {
+        if options.web3AuthClientId.isEmpty {
             throw CoreKitError.invalidInput
         }
-        
-        option = CoreKitOptions(disableHashFactor: disableHashFactor, Web3AuthClientId: web3AuthClientId, network: web3AuthNetwork, manualSync: manualSync) // TODO: This could be passed in instead
-        
-        appState = CoreKitAppState()
-
-        network = web3AuthNetwork
+        option = options
 
-        torusUtils = TorusUtils(enableOneKey: true, network: network.toTorusNetwork(), clientId: web3AuthClientId)
+        nodeDetailsManager = NodeDetailManager(network: option.web3AuthNetwork)
+        let torusOptions = TorusOptions(clientId: option.web3AuthClientId, network: option.web3AuthNetwork, enableOneKey: true)
+        torusUtils = try TorusUtils(params: torusOptions)
 
-        nodeDetailsManager = NodeDetailManager(network: network.toTorusNetwork())
+        coreKitStorage = CoreKitStorage(storeKey: storeKey, storage: option.localStorage)
 
-        coreKitStorage = CoreKitStorage(storeKey: storeKey, storage: localStorage)
-    }
-
-    public func updateAppState(state: CoreKitAppState) async throws {
-        appState.merge(with: state)
-
-        let jsonState = try JSONEncoder().encode(appState).bytes
-        try await coreKitStorage.set(key: localAppStateKey, payload: jsonState)
+        let config = CustomAuthArgs(urlScheme: "tdsdk://tdsdk/oauthCallback", network: option.web3AuthNetwork, enableOneKey: true, web3AuthClientId: option.web3AuthClientId)
+        customAuth = try CustomAuth(config: config)
+        userInfo = nil
     }
 
     public func getCurrentFactorKey() throws -> String {
```

Comment:
in web this function also returns shareType (index) for factor key along with factor key

Reply:
can be done @ieow